Author: Ken Zhu <guigenz@codeaurora.org>
Date:   Wed Dec 2 17:06:59 2020 -0800

    netifd: fix ip lost after dhcp renew
    
    After dhcp renew, the deletion of the old ip could removed the new ip.
    Delete the old ip before assigning the new ip to the device.

    Change-Id: I5378d403117348082a20483f87461103ffcdc6ac
    Signed-off-by: Ken Zhu <guigenz@codeaurora.org>

diff -Nur a/proto-shell.c netifd-2015-12-16/proto-shell.c
--- a/proto-shell.c	2020-12-02 17:03:10.718120624 -0800
+++ netifd-2015-12-16/proto-shell.c	2020-12-02 17:03:51.309929839 -0800
@@ -541,6 +541,7 @@
 		device_set_present(dev, true);
 
 		interface_update_start(iface);
+		interface_update_complete(state->proto.iface);
 	}
 
 	proto_apply_ip_settings(iface, data, addr_ext);
@@ -560,8 +561,6 @@
 	if ((cur = tb[NOTIFY_DATA]))
 		proto_shell_parse_data(state->proto.iface, cur);
 
-	interface_update_complete(state->proto.iface);
-
 	if ((state->sm != S_SETUP_ABORT) && (state->sm != S_TEARDOWN)) {
 		if (!keep)
 			state->proto.proto_event(&state->proto, IFPEV_UP);
